feature/185 글쓰기 버튼 클릭 후 뒤로가기 버튼이 페이지를 벗어나지 않도록 한다.

hwookim opened the pull request:
resolve #185 

/post-create 가 추가되었습니다.
이제 브라우저 내의 뒤로가기 하면 기본 화면으로 이동하며 맵의 모드가 바뀝니다.

추가적으로 처음 화면 돌입 시 "/"이 아니라 다른 경로로 들어오면 "/"로 리다이렉팅 하도록 변경했어요,

hwookim committed:
feature/185 feat: router 경로 post-create 추가

 - 마커 모드, 글 작성 모드 시 /post-create로 이동
 - 글 작성 모드 종료 시 경로 복귀

hwookim committed:
feature/185 feat: "/" 로 이동 시 MAP_MODE.DEFAULT로 변경

 - meta data postCreate 추가
   - "/post-create"에서 true
 - watch를 통해 postCreate에 따라 MAP_MODE 변경

hwookim committed:
feature/185 feat: PostCreateModal 종료 트랜지션 적용

hwookim committed:
feature/185 feat: cancelButton 삭제 후 backButton으로 대체

hwookim committed:
feature/185 feat: 시작을 "/"에서 하지 않으면 "/"로 리다이렉팅

LTTTTTE reviewed (approved):
브라우저 뒤로가기도 잘 catch하네요
이상한 url로 들어갔을때 항상 root로 가는 코드도 👍🏼

신기한게 $router.beforEach 인자중에 from을 사용안하는데도 eslint가 안잡아내네요

yebink commented on front/src/components/map/PostCreateModal.vue:
```diff
@@ -50,8 +50,19 @@ export default {
       location: this.$kakaoMap.getCenterLocation(),
       buttonColor: DORAN_DORAN_COLORS.POINT_COLOR,
       rendered: false,
+      flag: false,
```
무엇을 의미하는 `flag`인지 좀더 구체적으로 작성해볼 수 있을 것 같아요!

hwookim replied:
트랜지션이 끝났는지 확인하는 변수라 `isTransitionDone`으로 바꾸었습니다.
잘한 네이밍인지 잘 모르겠네요..🤔

yebink commented on front/src/router/index.js:
```diff
@@ -13,10 +13,18 @@ const routes = [
     },
     children: [
+      {
+        path: "post-create",
```
이러한 path들을 상수로 한 번에 관리해보는 건 어떨까요?
"/", "post-create" 이런 식으로요!

hwookim replied:
`$router.push`를 통해 이동한다면 상수화된 친구들을 사용할 수 있겠지만,
라우터 이동의 기본 방법인 `router-link`태그의 to 속성을 이용할 때는 상수를 사용하지 못하겠네요...🤔
상수화를 진행하면 좋을 거 같기는 한데 to 속성 내에서는 어떻게 해야할까요.

hwookim replied:
아 `:to` 를 통해서 사용할 수 있겠네요.

hwookim replied:
머지되고 나면 #243 을 통해 해결해보도록 할게요~

yebink replied:
오호..! 한참 찾아보고 있었는데 이런 방법이 있었군요!
역시 똑부러지네요 💯

yebink commented on front/src/components/map/PostCreateModal.vue:
```diff
@@ -50,8 +50,19 @@ export default {
+      this.rendered = false;
+      next(false);
+    });
+    this.$once("hook:destroyed", preventRoute);
```
👍

yebink reviewed (approved):
👍 이제 GA에서 페이지별 분석이 조금 더 원활해지겠군요!
수고했어요!
몇 가지 작은 피드백 남겼습니다~

hwookim committed:
feature/185 refactor: 불분명한 변수명 flag 변경

 - 트랜지션이 끝났는지 확인하는 변수임
 - flag -> isTransitionDone

chws commented on front/src/App.vue:
```diff
@@ -49,6 +49,9 @@ export default {
     },
   },
   async created() {
+    if (this.$route.path !== "/") {
```
👍

chws reviewed (approved):
리뷰가 늦었네요 우😅
깔끔하며 잘 동작하네요. 수고하셨습니다~

YejiAhn commented on front/src/App.vue:
```diff
@@ -49,6 +49,9 @@ export default {
     },
   },
   async created() {
+    if (this.$route.path !== "/") {
```
/timeline 페이지나 /post-create 페이지에서 새로고침을 눌러도 루트페이지로 되돌아가는데
의도하신 걸까요?! 동작하는 페이지들은 제외하고 루트페이지로 가게 하는게 맞지 않을까 생각도 드네요..

(+) 기간이나 검색 필터링은 지금 url 자체로 동작하진 않는데 새로고침을 해도 같은 페이지를 볼 수 있게 하면 좋을 것 같아요.
나중에 회의에서 의논해보면 좋을 것 같군요..! 🙂

hwookim replied:
버튼을 이용한 라우터 이동이 아닌 post-create로 직접 페이지 요청을 하게된다면 작동이 꼬여서 메인 화면 이외의 모든 페이지 요청은 메인화면으로 리다이렉팅 하도록 했습니다.

서비스 내에서 제공하는 새로고침 버튼이 아니라 브라우저를 통한 새로고침을 이용한다면 지도의 중심 위치가 현재 위치로 바뀌게 됩니다.
그렇게 하려면 url에서 가지고 있어야 할 정보 중에 가장자리 4점의 위치까지 가지고 있어야겠네요.
현재는 같은 페이지를 보장할 수 없으니 메인페이지로 리다이렉트하는 편이 맞다고 봐요.🤔

다만 게시글 페이지는 리다이렉트나 직접 요청시에도 해당 게시글을 보여주는 것이 맞는 것 같아 수정하도록 했습니다.

YejiAhn reviewed (approved):
우 리뷰 늦어서 미안해요! 
잘 동작하는거 확인했습니다. 수고 많았어요 💯

hwookim committed:
feature/185 feat: 게시글 화면에서 리다이렉트 시 메인 페이지로 넘어가지 않도록 변경

 - App.checkUrl -> checkUrlToken
 - App.checkUrl은 최초 접근 url으로 맵페이지, 게시글페이지가 아닌 경우 맵페이지로 리다이렉트하는 메소드로 변경
